common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp

```diff
@@ -614,8 +669,28 @@ bool ap_wlan_hal_nl80211::process_nl80211_event(parsed_obj_map_t &parsed_obj)
     } break;
 
     // ACS/CSA Completed
+    case Event::CTRL_EventChannelSwitch: {
+        std::string bandwidth = parsed_obj["ch_width"];
+        if (bandwidth.empty()) {
+            LOG(ERROR) << "Invalid bandwidth";
+            return false;
+        }
+        m_radio_info.channel =
+            beerocks::utils::wifi_freq_to_channel(beerocks::string_utils::stoi(parsed_obj["freq"]));
+        m_radio_info.bandwidth          = wpa_bw_to_beerocks_bw(bandwidth);
+        m_radio_info.channel_ext_above  = beerocks::string_utils::stoi(parsed_obj["ch_offset"]);
+        m_radio_info.vht_center_freq    = beerocks::string_utils::stoi(parsed_obj["cf1"]);
+        m_radio_info.is_dfs_channel     = beerocks::string_utils::stoi(parsed_obj["dfs"]);
+        m_radio_info.last_csa_sw_reason = ChanSwReason::Unknown;
+        if (son::wireless_utils::which_freq(m_radio_info.channel) == beerocks::eFreqType::FREQ_5G) {
+            m_radio_info.is_5ghz = true;
+        }
+    } break;
     case Event::ACS_Completed:
+        break;
```

Comment:
Why did you add a break here? It means we won't push an event for ACS completed.

Reply:
Done